src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -173,7 +245,26 @@ internal set
         /// <summary>
         /// Gets or sets the telemetry channel.
         /// </summary>
-        public ITelemetryChannel TelemetryChannel { get; set; }
+        public ITelemetryChannel TelemetryChannel
+        {
+            get
+            {
+                return this.telemetryChannel;
+            }
+
+            set
+            {
+                ITelemetryChannel oldChannel = this.telemetryChannel;
+                this.telemetryChannel = value;
+
+                // If we have a previously assigned channel which is not the same one as the "new" value
+                // passed in then we need to dispose of the old channel to keep from leaking resources.
+                if (oldChannel != null && oldChannel != value)
+                {
+                    oldChannel.Dispose();
```

Comment:
I was afraid `StreamReader` would come up 😭 It's such a horrible anti-pattern. 

I'm a bit rusty on lifetime management of singletons in the SDK. 
- Could you elaborate on .NET core registering everything as singletons? Does it mean that the service container disposes them at application shutdown?
- How does `TelemetryConfiguration.Active` get disposed on other .NET platforms?